docker/Dockerfile

```
@@ -0,0 +1,291 @@
+# This Dockerfile can be built from scratch with e.g the command
+# docker build . -t my_numba_dpex_build
+# The ARGs in the header of this file can be changed as needed (either by
+# editing the file or with the command line).
+# The build can be fairly long (up to one hour on low-end machines.)
+# To start a container, use e.g
+# sudo docker run --rm --name my_container -it -v /my/host/volume/:/mounted/volume --device=/dev/dri my_numba_dpex_build
+# The parameter --device=/dev/dri enables gpu passthrough (as long as the user 
+# has permissions, e.g by being a member of the video group)
+
+
+# TODO: make a custom build of the base image to test with other base OS
+# (e.g ubuntu2204)
+# ???: actually not sure if it's needed or interesting to use the sycl nightly
+# binaries at all. The nightly also ship an up-to-date installation of the
+# intel runtime, including drivers and opencl and level_zero runtimes.
+
+ARG BASE=ghcr.io/intel/llvm/sycl_ubuntu2004_nightly
+ARG TMPDIR=/tmp
+
+# Various resources about the SYCL toolchain
+# Environment variables with important effects on the SYCL toolchain:
+# https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#sycl_device_filter
+# Overview of the SYCL build instructions and environment requirements: https://intel.github.io/llvm-docs/GetStartedGuide.html
+# More about sycl docker image Dockerfiles: https://github.com/intel/llvm/tree/sycl/devops/containers
+# and https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/DockerBKMs.md
+
+
+ARG INTEL_PYPI_URL=https://pypi.anaconda.org/intel/simple
+ARG BASE_PYPI_URL=https://pypi.org/simple
+
+# /!\ the following version must exist in intel pypi. It is recommended to
+# not use version ranges, because if a higher version exists in pypi.org it
+# will be preferred over the versions in the intel pypi.
+
+ARG INTEL_NUMPY_VERSION="==1.22.3"
```

This should resolve [the problem mentioned by @ogrisel](https://github.com/ogrisel/sklearn-numba-dpex/pull/1/files#r914945843)
```suggestion
ARG INTEL_NUMPY_VERSION="1.22.3"
```